### PR TITLE
cli: add guest hook path option

### PR DESCRIPTION
**Pull request by eguzman3**
Blocked on (https://github.com/kata-containers/agent/pull/365), will need to revendor.

### CODE REVIEW

**Comment by jodh-intel** [comment]
Added `do-not-merge` label which will disallow this branch from landing - see `.pullapprove.yml`.

Once the dependent branch is merged, we can remove the label ;)

File: cli/config.go [hunk 1]
```diff
@@ -275,6 +276,13 @@ func (h hypervisor) useVSock() bool {
 	return h.UseVSock
 }
 
+func (h hypervisor) guestHookPath() string {
```
Comment (jodh-intel) [inline cli/config.go]: Please could you add a minimal test in `config_test.go` for this. Looking around your change, it seems we have been a little remiss recently as some of the newer calls are missing unit tests.

You could re-use code from `TestHypervisorDefaultsImage()` - we just need to test the two scenarios - either the default value is returned or the `h.GuestHookPath`.

**Comment by egernst** [comment]
Hey @eguzman3  - it seems like this PR has more than just adding the guest hooks - can you split into multiple well described commits, and perhaps as seperate PRs?  Please let me know if I'm mistaken...

**Comment by egernst** [comment]
And, please disregard my comment -- I see that I was in the weeds looking at the vendor code.  Thanks!

**Comment by sboeuf** [comment]
@egernst @eguzman3 once kata-containers/agent#365 lands, it's still cleaner if you split the vendoring code in its own commit ;)

**Comment by jodh-intel** [comment]
Still blocked on https://github.com/kata-containers/agent/pull/365.

**Comment by raravena80** [comment]
@eguzman3 ping (From your weekly Kata herder). It is still blocked?

**Comment by caoruidong** [comment]
I think we can drop this one cause there is a new version #834

**Pull request closed without merging**